web/ui/src/Pools/PoolsController.js

```diff
@@ -6,33 +6,110 @@
 (function() {
     'use strict';
 
-    controlplane.controller("PoolsController", ["$scope", "$routeParams",
-    "resourcesFactory", "authService", "$modalService", "$translate",
-    "$notification", "areUIReady", "$interval", "servicedConfig", "log",
-    "POOL_PERMISSIONS", "miscUtils", "Pool",
-    function($scope, $routeParams, resourcesFactory, authService, $modalService,
-    $translate, $notification, areUIReady, $interval, servicedConfig, log,
-    POOL_PERMISSIONS, utils, Pool){
-
-        // Ensure logged in
-        authService.checkLogin($scope);
-
-        // allow templates to get the list
-        // of permissions
-        $scope.permissions = POOL_PERMISSIONS;
-
-        $scope.clickPool = function(id) {
-            resourcesFactory.routeToPool(id);
-        };
-
-        // Function to remove a pool
-        $scope.clickRemovePool = function(poolID) {
-            if ($scope.isDefaultPool(poolID)) {
-              return;
+    class PoolsController {
+
+        constructor ($scope, resourcesFactory, authService, $modalService, 
+                     $translate, $notification, areUIReady, $interval, servicedConfig, 
+                     log, POOL_PERMISSIONS, miscUtils, Pool, $q)
+        {
+            authService.checkLogin(this);
+
+            this.resourcesFactory = resourcesFactory;
+            this.$modalService = $modalService;
+            this.$translate = $translate;
+            this.$notification = $notification;
+            this.areUIReady = areUIReady;
+            this.$interval = $interval;
+            this.utils = miscUtils;
+            this.$q = $q;
+            this.permissions = POOL_PERMISSIONS;
+            this.name = "pools";
+
+            this.refreshPools().then(() => $scope.$emit("ready"));
+
+            this.updateFrequency = 3000;
+            servicedConfig.getConfig()
+                .then(config => {
+                    this.updateFrequency = config.PollFrequency * 1000;
+                }).catch(err => {
+                    let errMessage = err.data ? err.data.Detail : err.statusText;
+                    log.error("could not load serviced config:", errMessage);
+                });
+
+            $scope.poolsTable = {
+                sorting: { id: "asc" },
+                watchExpression: () => this.lastUpdate
+            };
+
+            $scope.breadcrumbs = [
+                { label: 'breadcrumb_pools', itemClass: 'active' }
+            ];
+
+            this.startPolling();
+
+            $scope.$on("$destroy", () => this.stopPolling());
+
+            this.newScope = () => $scope.$new(true);
+            this.newPool = (data) => new Pool(data);
```

Comment:
is `newPool` necessary? If so, it can be a regular method and not defined in the constructor. However, Id say just to inline it in the one place it is used.